x-pack/plugins/alerting/server/routes/snooze_rule.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IRouter } from 'kibana/server';
+import { schema } from '@kbn/config-schema';
+import { ILicenseState, AlertTypeDisabledError } from '../lib';
+import { RewriteRequestCase, verifyAccessAndContext } from './lib';
+import { AlertingRequestHandlerContext, BASE_ALERTING_API_PATH } from '../types';
+
+const paramSchema = schema.object({
+  id: schema.string(),
+});
+
+const bodySchema = schema.object({
+  snooze_end_time: schema.maybe(schema.string()),
+});
+
+const rewriteBodyReq: RewriteRequestCase<{
+  snoozeEndTime?: string;
+}> = ({ snooze_end_time: snoozeEndTime }) => ({
+  snoozeEndTime,
+});
+
+export const snoozeRuleRoute = (
+  router: IRouter<AlertingRequestHandlerContext>,
+  licenseState: ILicenseState
+) => {
+  router.post(
+    {
+      path: `${BASE_ALERTING_API_PATH}/rule/{id}/_snooze`,
```

Comment:
We want this api to be internal since we are in the MVP phase
```suggestion
      path: `${INTERNAL_BASE_ALERTING_API_PATH}/rule/{id}/_snooze`,
```

Reply:
Also make sure to add the `snooze_end_time` attribute [here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/alerting/server/rules_client/rules_client.ts#L261-L264)

<img width="681" alt="image" src="https://user-images.githubusercontent.com/189600/157293514-679f8260-9d6d-4ba0-a776-7a887ca30a96.png">